### DISCUSSION TITLE
connect to MS sql server with integrated using .net framework (c#)

### DISCUSSION THREAD

**Discussion by Abhisheik-athi** [dotnet/SqlClient#1486] [Q&A]
i am using .net framework and trying to connect to a MS SQL server with integrated security = true. Need to use different credentials so is there a way using kerberos authentication. and authenticate the creds that is needed for the sql connection ?
need help on how to do it.

**Comment by JRahnama** [comment]
@Abhisheik-athi is it a windows based or linux?

**Comment by Abhisheik-athi** [comment]
windows . sql server is in a on prem server ..

**Comment by JRahnama** [comment]
Do you have your kerberos and SP setup and tested? I think you might be able to authenticate that way. Check with `SELECT auth_scheme from sys.dm_exec_connections where session_id = @@spid` and it should be showing `KERBEROS` and not `NTLM`

**Reply by Abhisheik-athi** [reply]
its NTLM.. i do not have access to SQL server though

**Comment by JRahnama** [comment]
Let me search on this and will get back to you shortly.

**Reply by Abhisheik-athi** [reply]
my ask is simple .. i have been given a jbdc url (integrated security with service account and password) ... 
i have build a console app and i am trying to establish the connection to that db but since the connection string has integration security true it takes the windows cred (which is my personal account and not the service account that i want to connect with)

**Comment by JRahnama** [comment]
If your machine is not registered with authentication server no, you cannot use kerberos. plus if the issue is with JDBC you are in a wrong repo. JDBC related issues go [here](https://github.com/Microsoft/mssql-jdbc)

**Reply by Abhisheik-athi** [reply]
issue is not with jdbc ... i have a question on the sql client (to build an app)

**Comment by JRahnama** [comment]
username and password authentication for Kerberos is not supported in ADO.NET. It only works for Integrated security =true

**Reply by Abhisheik-athi** [reply]
when creating a connection with ms sql from a .net based app there is an option to pass AccessToken. How to get ntlm based token and pass that to connection.AccessToken property

**Reply by Abhisheik-athi** [reply]
integrated security is true the .net console app is not running with credentials that has permission to access the DB.. what do i do in this case. I need to make the connection work with the different creds and believe i need to pass through as a token (ntlm based).
